tests/performance/local/foreach_scaling.cpp

```diff
@@ -71,6 +71,24 @@ namespace hpx { namespace parallel { namespace execution {
     };
 }}}    // namespace hpx::parallel::execution
 
+///////////////////////////////////////////////////////////////////////////////
+void measure_plain_for(std::vector<std::size_t> const& data_representation)
+{
+    std::size_t size = data_representation.size();
+    for (std::size_t i = 0; i != size; ++i)
+    {
+        worker_timed(delay);
+    }
```

Comment:
Might be worth trying? 
```suggestion
    while (size >= 4)
    {
        worker_timed(delay);
        worker_timed(delay);
        worker_timed(delay);
        worker_timed(delay);
        size -= 4;
    }
    // handle the rest like in the hpx for loops
```

Reply:
Yes, true - this will bring things on par. Good catch.

Reply:
That's changed now and indeed the plain `for` loop is as fast as our algorithm, as expected:
```
----------------Parameters---------------------
Vector size                       : 10000000
Number of tests                   :       10
Delay per iteration (nanoseconds) :        0
Display time in                   : Seconds
-------------Average-(for)---------------------
Average execution time (unrolled) : 0.00547917
Average execution time (iter)     : 0.00877946
-------------Average-(for_each)----------------
Average parallel execution time   : 0.00158009
Average task execution time       : 0.00069507
Average sequential execution time : 0.00485273
-----Execution Time Difference-(for_each)------
Parallel Scale                    :  3.07117
Task Scale                        :  6.98164
-------------Average-(for_loop)----------------
Average parallel execution time   : 0.00060268
Average task execution time       : 0.000612
Average sequential execution time : 0.00490469
-----Execution Time Difference-(for_loop)------
Parallel Scale                    :  8.13813
Task Scale                        :   8.0142
```

Reply:
@hkaiser  Sorry for bringing up this late, but I think I am missing something: why is "Average parallel execution time" for `for_each` bigger than your initial (improved) measurements? Seems like `for_each` Parallel Scale is doubled down...

Reply:
Good question, let me check this.